test/python/circuit/test_controlled_gate.py

```diff
@@ -792,7 +792,12 @@ def test_base_gate_setting(self, gate_class):
 
         base_gate = gate_class(*free_params)
         cgate = base_gate.control()
-        self.assertEqual(base_gate.base_gate, cgate.base_gate)
+
+        # the base gate of CU is U (3 params), the base gate of CCU is CU (4 params)
+        if gate_class == CUGate:
+            self.assertListEqual(cgate.base_gate.params[:3], base_gate.base_gate.params[:3])
+        else:
+            self.assertEqual(base_gate.base_gate, cgate.base_gate)
```

Comment:
@ewinston I updated this test to account for the different parameters in the base gate of CU and CCU, if this is fine with you feel free to merge!

Reply:
ok, thanks!